xbmc/ServiceManager.h

```diff
@@ -168,4 +177,7 @@ class CServiceManager
   std::unique_ptr<CPlayerCoreFactory> m_playerCoreFactory;
   std::unique_ptr<CDatabaseManager> m_databaseManager;
   std::unique_ptr<CMediaManager> m_mediaManager;
+#if !defined(TARGET_WINDOWS) && defined(HAS_DVD_DRIVE)
+  std::unique_ptr<MEDIA_DETECT::CDetectDVDMedia> m_DetectDVDType;
```

Comment:
From one dumping ground to another. It would be a good time to consider a new home for anything here, but not required for the PR.

Reply:
From what I've gathered, servicebroker was just a stepping stone to greater things which just haven't materialised. I still think it helps to get everything here to give a clear picture of what to do with the DI stuff.
I've tried comprehending jimfcarrols DI stuff, but it hasn't sunk in how I could do it as yet. Maybe one day, haha

Reply:
@jimfcarroll's DI stuff improves our tangle of dependencies by replacing imperative relationships with declarative relationships. What we truly need is a decoupling of dependencies. However, it's a shame that "how it all should be" got in the way of the tangible benefits of DI for the parts that don't need decoupling.